core/amber/src/main/scala/edu/uci/ics/amber/engine/common/client/AmberClient.scala

```diff
@@ -59,7 +64,7 @@ class AmberClient(system: ActorSystem, workflow: Workflow, controllerConfig: Con
     }
   }
 
-  def getObservable[T](implicit ct: ClassTag[T]): Observable[T] = {
+  def registerCallback[T](callback: T => Unit)(implicit ct: ClassTag[T]): Subscription = {
```

Comment:
does the call back have to return `Unit`?

Reply:
Yes, it has to.